### PR TITLE
Guess proxy no proxy

### PR DESCRIPTION
**Pull request by amirbilu**

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -26,19 +28,43 @@ function noProxy(selfHostedServerUrl: string): boolean {
...
+    return false;
+  }
+}
+export default async function client(
+  selfHostedServerUrl: string
+): Promise<AxiosInstance> {
+  const axiosClient = axios.create({
+    baseURL: selfHostedServerUrl,
+    httpAgent: new http.Agent(),
+    httpsAgent: new https.Agent(),
```
Comment (amirbilu) [inline src/client.ts:46]: This is since vscode overrides the default agent with proxy configuration

File: src/updateTask.ts [hunk 2]
```diff
@@ -32,7 +32,7 @@ export default async function updateTask(
         const path = await createTmpFile();
         await downloadUrl(
           client,
-          `${UPDATE_PREFIX}/tabnine-vscode-${latestVersion.trim()}.vsix`,
```
Comment (yanir-codota) [inline src/updateTask.ts:35]: why?

Comment (amirbilu) [reply src/updateTask.ts:35]: Already trimmed at line 22

File: src/client.ts [hunk 3]
```diff
@@ -26,19 +28,43 @@ function noProxy(selfHostedServerUrl: string): boolean {
...
+    proxy: false,
+  });
   const url = proxyUrl();
+  if (!url) {
+    return axiosClient;
+  }
+
+  const { hostname: host, protocol, port } = Url.parse(url);
+  if (!host || !protocol || !port || noProxy(selfHostedServerUrl)) {
```
Comment (yanir-codota) [inline src/client.ts:55]: does it always have all? port is required?

Comment (amirbilu) [reply src/client.ts:55]: probably not but most of the proxies are not 80/443 so let's avoid of another code atm

**Review by yanir-codota — APPROVED** [review]
LGTM

File: src/client.ts [hunk 4]
```diff
@@ -26,19 +28,43 @@ function noProxy(selfHostedServerUrl: string): boolean {
...
+    return axiosClient;
+  }
+
+  const axiosClientWithProxy = axios.create({
+    baseURL: selfHostedServerUrl,
+    httpAgent: new http.Agent(),
```
Comment (assapir) [inline src/client.ts:61]: Do we want to pass in some other options?

**Review by assapir — APPROVED** [review]
Proxy proxy java java

**Comment by yanir-codota** [comment]
> Proxy proxy java java

הפוך 😂

**Pull request merged**